src/Chainweb/Rosetta/RestAPI/Server.hs

```diff
@@ -0,0 +1,121 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+-- |
+-- Module: Chainweb.Rosetta.RestAPI.Server
+-- Copyright: Copyright © 2018 - 2020 Kadena LLC.
+-- License: MIT
+-- Maintainer: Colin Woodbury <colin@kadena.io>
+-- Stability: experimental
+--
+--
+module Chainweb.Rosetta.RestAPI.Server where
+
+import Control.Error.Util
+import Control.Monad.IO.Class
+import Control.Monad.Trans.Except
+
+import qualified Data.ByteString.Short as BSS
+import Data.Proxy (Proxy(..))
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Data.Vector as V
+
+import Rosetta
+
+import Servant.API
+import Servant.Server
+
+import Text.Read (readMaybe)
+
+-- internal modules
+
+import Chainweb.Mempool.Mempool
+import Chainweb.RestAPI.Utils (SomeServer(..))
+import Chainweb.Rosetta.RestAPI
+import Chainweb.Version
+
+---
+
+rosettaServer
+    :: forall a (v :: ChainwebVersionT)
+    . [(ChainId, MempoolBackend a)]
+    -> Server (RosettaApi v)
+rosettaServer ms = (const $ error "not yet implemented")
```

Comment:
So if we need more backend components we add it to the list here?

Reply:
Yes, alongside the `ms`.